src/libraries/token/VariableSupplyToken.sol

```diff
@@ -0,0 +1,23 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/libraries/token/StandardToken.sol';
+
+
+contract VariableSupplyToken is StandardToken {
+    using SafeMath for uint256;
+
+    event Mint(address indexed target, uint256 value);
+    event Burn(address indexed target, uint256 value);
+
+    function mint(address _target, uint256 _amount) internal returns (bool) {
```

Comment:
Ah, that is a separate thing unrelated to this.  This is just a primitive for tokens that need the ability to mint/burn.  The thing you are thinking about is a strategy that would (if/when implemented) leverage this primitive.